openstack/client.go

```diff
@@ -107,6 +108,10 @@ func v2auth(client *gophercloud.ProviderClient, endpoint string, options gopherc
 		return err
 	}
 
+	if options.AllowReauth {
+		client.ReauthFunc = AuthenticateV2
+		client.AuthOptions = options
+	}
```

Comment:
You could also capture `AuthOptions` (and the client pointer) with a closure, instead:

```go
client.ReauthFunc = func() error {
  return AuthenticateV2(client, options)
}
```

Pros:
 * Cleaner (in my opinion, anyway).
 * Avoids unnecessarily exporting `AuthOptions` in the client struct.
 * Can capture more context in the future if we ever need it without breaking the function's type signature and therefore backwards compatibility (although we'd most likely need to add it to `AuthOptions` instead).

Cons:
 * A little more magical